cache/redis_client.go

```diff
@@ -110,7 +110,7 @@ func (r *RedisCache) Remove(ctx context.Context, key string, field string) {
 func (r *RedisCache) HealthCheck(ctx context.Context) error {
 	res := r.client.Ping(ctx)
 	if res.Err() != nil {
-		return fmt.Errorf("redis failed to respond")
+		return fmt.Errorf("redis failed to respond err: %s", res.Err())
```

Comment:
Is Err() a error or string? If its an error %w might be a good option to wrap it, if we then wanted to assert the type of error higher up.